### PR TITLE
add __await__ for Pipeline class

### PR DESCRIPTION
**Pull request by stanislau-arkhipenka**
### TLDR:
without the fix this example:
```
my_key: str = "test_1"
async with redis_client.pipeline() as pipe:
    await pipe.watch(my_key)
    pipe.multi()
    await pipe.set(my_key, 0)
    await pipe.execute()
```
fails with something like this:
```
....
  File "aioredis/client.py", line 879, in initialize
AttributeError: 'Pipeline' object has no attribute 'single_connection_client'
```
### Little bit more details:
The pipeline (pipe) is redis-py's implementation of redis transaction (https://redis.io/topics/transactions).
In terms of aioredis implementation:
`Pipeline `is a child class of `Redis `class
When you try run `pipe.set()` you actually run method from parent (Redis)
inside of this parent's method (pipe.set()) special function `execute_command()` is executed 
This `execute_command()` is overridden in child `Pipeline `and looks something like this:
```
def execute_command(bla):
        if bla:
            return self.immediate_execute_command()
        return self.pipeline_execute_command()
```
The problem, is that `immediate_execute_command `is async function (async def ...) and `pipeline_execute_command `is NOT (def ...).
Developers didn't wanted to have inconsistancy in returns (sometimes awaitable, sometimes just value), so decided to make `pipeline_execute_command `to return self, so when you execute await `pipe.set() `you actually execute _await_. But await is not defined in child class (Pipeline), therefor parent's (Redis's) method executed, which in turn `executes self.initialize()` which reference `self.single_connection_client` which is not defined in child subclass (Pipeline).

### CODE REVIEW

**Comment by Andrew-Chen-Wang** [comment]
@stanislau-arkhipenka is this a duplicate of #1027?

**Comment by stanislau-arkhipenka** [comment]
@Andrew-Chen-Wang yeap, seems this solves pretty much same issue. 
In #1027 author sets single_connection_client to False, so no attribute error will occure, when initialze is called. I don't particularry like it as Pipeline should not have a context on connection, as it operates only with connection pool.

Nevertheless both solutions will work :)

**Comment by rad-pat** [comment]
@stanislau-arkhipenka, I think your solution is better myself. Perhaps you can use the test from my https://github.com/aio-libs/aioredis-py/pull/1027 PR to cover the code?

**Comment by Andrew-Chen-Wang** [comment]
Awesome. Thanks for checking out each other's PR. @stanislau-arkhipenka adding the other PR's test would be great!

**Comment by stanislau-arkhipenka** [comment]
Also added other PR to CONTRIBUTORS as I basically ctrl+c ctrl+v the test

**Review by seandstewart — CHANGES_REQUESTED** [review]
Lets make `async_self` -> `_async_self` since it's an implementation detail, not an official API.

**Review by seandstewart — APPROVED** [review]
LGTM, thanks for making that change. Go ahead and [add a news artifact](https://aioredis.readthedocs.io/en/latest/devel/#getting-started) to the CHANGES directory and then I think this is good to go.
